supplementary/alica_ros1/alica_ros_turtlesim/libalica-turtlesim/src/FourCorners.cpp

```diff
@@ -0,0 +1,41 @@
+#include "FourCorners.h"
+#include "FourCornersSummand.h"
+#include "engine/USummand.h"
+#include "engine/blackboard/Blackboard.h"
+#include "ros/ros.h"
+namespace turtlesim
+{
+
+std::shared_ptr<FourCornersUtilityFunction> FourCornersUtilityFunction::create(alica::UtilityFunctionContext& context)
+{
+    return std::make_shared<FourCornersUtilityFunction>(context);
+}
+
+std::shared_ptr<alica::UtilityFunction> FourCornersUtilityFunction::getUtilityFunction(alica::Plan* plan)
+{
+    // contribution weightage in utilty score for similar assignments
+    double similarityWeight = 0.5;
+    // in this case priorityWeight is 0 as all agents role are same.
+    // it only comes into account when the role of agent differs and we have priority of each task based on agent role
+    double priorityWeight = 0;
+    double customWeight = 0.5;
+    std::shared_ptr<alica::UtilityFunction> function = std::make_shared<alica::UtilityFunction>(priorityWeight, similarityWeight, plan);
+    FourCornersSummand* summand = new FourCornersSummand(customWeight);
+    summand->addEntryPoint(plan->getEntryPointByID(599427197773116147));
```

Comment:
is it possible to get by name?

Reply:
I couldn't find any method to get this my name

Reply:
maybe comment the name after each of these lines, I have no idea what these numbers mean